app/src/main/java/org/p2p/wallet/auth/interactor/restore/CustomShareRestoreInteractor.kt

```diff
@@ -0,0 +1,81 @@
+package org.p2p.wallet.auth.interactor.restore
+
+import com.google.gson.Gson
+import com.google.gson.JsonObject
+import org.p2p.solanaj.utils.crypto.decodeFromBase64
+import org.p2p.wallet.auth.gateway.api.request.OtpMethod
+import org.p2p.wallet.auth.gateway.api.response.ConfirmRestoreWalletResponse
+import org.p2p.wallet.auth.gateway.repository.GatewayServiceRepository
+import org.p2p.wallet.auth.repository.RestoreFlowDataLocalRepository
+import org.p2p.wallet.auth.web3authsdk.response.Web3AuthSignUpResponse
+import org.p2p.wallet.utils.fromJsonReified
+import timber.log.Timber
+
+class CustomShareRestoreInteractor(
+    private val gatewayServiceRepository: GatewayServiceRepository,
+    private val restoreFlowDataLocalRepository: RestoreFlowDataLocalRepository,
+    private val gson: Gson,
+) {
+    class RestoreWalletFailure(override val message: String) : Throwable(message)
+
+    suspend fun startRestoreCustomShare(userPhoneNumber: String) {
+        val temporaryUserPublicKey = restoreFlowDataLocalRepository.userRestorePublicKey
+            ?: throw RestoreWalletFailure("User restore public key is null")
+        val temporaryUserPrivateKey = restoreFlowDataLocalRepository.userRestorePrivateKey
+            ?: throw RestoreWalletFailure("User restore private key is null")
+
+        gatewayServiceRepository.restoreWallet(
```

Comment:
you forgot to check for lastEnteredNumber which stored in repository with new entered phone number, if they are eqauls, you can skip sending request